File: Dataset/Leetcode/test/111/559.py
class Solution:
    def XXX(self, root: TreeNode) -> int:
        # BFS，遇左右节点均为空打印退出
        if not root:
            return 0
        res = []
        queque = [(1, root)]
        while queque:
            depth, first = queque.pop(0)
            if not first.left and not first.right:
                return depth
            if first.left:
                queque.append((depth+1, first.left))
            if first.right:
                queque.append((depth+1, first.right))

File: Dataset/Leetcode/test/101/559.java
 class Solution {
    public boolean XXX(TreeNode root) {
        if(root==null)
            return true;
        return isS(root.left,root.right);
    }

    public boolean isS(TreeNode left,TreeNode right){
        if(left==right)
            return true;
        if(left==null||right==null)
            return  false;
        return left.val==right.val&&isS(left.left,right.right)&&isS(left.right,right.left);
    }
}

